packages/frontend/src/pages/scaling/costs/props/getScalingCostsView.ts

```diff
@@ -0,0 +1,162 @@
+import { Layer2 } from '@l2beat/config'
+import {
+  assert,
+  L2CostsApiProject,
+  L2CostsApiResponse,
+  L2CostsDetails,
+  notUndefined,
+} from '@l2beat/shared-pure'
+
+import { formatLargeNumber } from '../../../../utils'
+import { formatCurrency } from '../../../../utils/format'
+import { orderByTvl } from '../../../../utils/orderByTvl'
+import {
+  CostsData,
+  CostsDataDetails,
+  CostsPagesData,
+  ScalingCostsViewEntry,
+} from '../types'
+import { ScalingCostsViewProps } from '../view/ScalingCostsView'
+
+export function getScalingCostsView(
+  projects: Layer2[],
+  pagesData: CostsPagesData,
+): ScalingCostsViewProps {
+  const { tvlApiResponse, costsApiResponse } = pagesData
+
+  const includedProjects = getIncludedProjects(projects, costsApiResponse)
+  const orderedProjects = orderByTvl(includedProjects, tvlApiResponse)
+
+  return {
+    items: orderedProjects
+      .map((project) => {
+        const l2CostsProjectData =
+          costsApiResponse.projects[project.id.toString()]
+        assert(l2CostsProjectData, 'l2CostsProjectData is undefined')
+
+        return getScalingCostsViewEntry(project, l2CostsProjectData)
+      })
+      .filter(notUndefined),
+  }
+}
+
+function getScalingCostsViewEntry(
+  project: Layer2,
+  l2CostsProjectData: L2CostsApiProject,
+): ScalingCostsViewEntry {
+  return {
+    name: project.display.name,
+    shortName: project.display.shortName,
+    slug: project.display.slug,
+    warning: project.display.warning,
+    redWarning: project.display.redWarning,
+    showProjectUnderReview: !!project.isUnderReview,
+    category: project.display.category,
+    provider: project.display.provider,
+    purposes: project.display.purposes,
+    stage: project.stage,
+    costs: getCostsData(l2CostsProjectData),
+  }
+}
```

Comment:
I guess we should also add new `hasImplementationChanged` property to that subpage